modules/tinymce/src/core/main/ts/html/Base64Uris.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Id, Obj } from '@ephox/katamari';
+
+export interface Base64UriMatch {
+  mime: string;
+  base64: string;
+}
+
+export interface Base64Extract {
+  prefix: string;
+  uris: Record<string, Base64UriMatch>;
+  html: string;
+}
+
+export const extractBase64DataUris = (html: string): Base64Extract => {
+  const dataImageUri = /data:(image\/[a-z]+);base64,([a-z0-9\+\/=]+)/gi;
```

Comment:
This regex won't work for images that have a `+` in the mime type, eg: `image/svg+xml` or other valid types such as the ones listed here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#Image_types

- [ ] Ensure this works with mime types that have a plus, hyphen, etc...